Content.Server/Fax/FaxSystem.cs

```diff
@@ -325,14 +347,35 @@ private void UpdateAppearance(EntityUid uid, FaxMachineComponent? component = nu
         if (!Resolve(uid, ref component))
             return;
 
+        if (!TryComp<FaxableObjectComponent>(component.PaperSlot.Item, out var faxable))
+            return;
+
+        if (component.PaperSlot.Item == null)
+            return;
+
         if (component.InsertingTimeRemaining > 0)
-            _appearanceSystem.SetData(uid, FaxMachineVisuals.VisualState, FaxMachineVisualState.Inserting);
+        {
+            switch (faxable.Identity)
+            {
+                case "hamster":
+                    _appearanceSystem.SetData(uid, FaxMachineVisuals.VisualState, FaxMachineVisualState.InsertingHamlet);
+                    break;
+                case "mouse":
+                    _appearanceSystem.SetData(uid, FaxMachineVisuals.VisualState, FaxMachineVisualState.InsertingMouse);
+                    break;
+                case "mothroach":
+                    _appearanceSystem.SetData(uid, FaxMachineVisuals.VisualState, FaxMachineVisualState.InsertingMothroach);
+                    break;
+                default:
+                    _appearanceSystem.SetData(uid, FaxMachineVisuals.VisualState, FaxMachineVisualState.Inserting);
+                    break;
+            }
+        }
         else if (component.PrintingTimeRemaining > 0)
```

Comment:
Tying this into code is still a bad idea as any changes need to touch C# when ss14 is supposed to be data-driven.

What if you just put it on its own appearance key + layer and then play the faxableobjects state if it has one.